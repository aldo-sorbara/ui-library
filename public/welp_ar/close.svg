<svg xmlns="http://www.w3.org/2000/svg"
    fill="currentColor"
    height="24"
    width="24"
    viewBox="0 0 35 35">
  <path d="M26 22.87A2.14 2.14 0 0 1 26 26a2.14 2.14 0 0 1-3.13 0l-5.54-5.54-5.5 5.5c-.45.45-.98.68-1.58.7A2.1 2.1 0 0 1 8.67 26 2.1 2.1 0 0 1 8 24.42c.01-.6.24-1.13.7-1.58l5.5-5.5-5.54-5.55a2.15 2.15 0 0 1 0-3.12 2.14 2.14 0 0 1 3.12 0l5.54 5.54 5.54-5.54a2.15 2.15 0 0 1 3.13 0 2.14 2.14 0 0 1 0 3.12l-5.54 5.54L26 22.87z"/>
</svg>
