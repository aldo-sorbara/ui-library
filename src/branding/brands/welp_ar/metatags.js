export const presto = {
  title: 'Presto Hoy Prestamos Online',
  description: 'Préstamo confiable, rápido y seguro',
  keywords: 'Préstamo, Crédito, Minicrédito, Presto Hoy, Efectivo, Cash'
};
export const posta = {
  title: 'Créditos Posta',
  description: 'Créditos Posta',
  keywords: 'Préstamo, Crédito, Posta, Efectivo, Cash'
};
export const welp = {
  title: 'W E L P',
  description: 'WELP',
  keywords: 'Préstamo, Crédito, Welp, Efectivo, Cash'
};
export default {
  title: 'Onboarding Title',
  description: 'Onboarding Description',
  keywords: 'Onboarding, React, Redux, Styled'
};
