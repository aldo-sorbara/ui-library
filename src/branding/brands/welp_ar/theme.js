import alerts from './jsons/alerts.json';
import assetsPaths from './jsons/assets.json';
import breakPoints from '../common/jsons/breakPoints.json';
import font from './jsons/font.json';
import palette from './jsons/palette.json';
import reset from './jsons/reset.json';
import spacer from './jsons/spacer.json';
import typography from './jsons/typography.json';
import zIndex from '../common/jsons/zIndex.json';

import buttons from './jsons/buttons.json';
import radioButton from './jsons/radioButton.json';
import list from './jsons/list.json';
import containers from './jsons/containers.json';
import modals from './jsons/modals.json';

export default {
  name: 'welp_ar',
  alerts,
  assetsPaths,
  breakPoints,
  buttons,
  list,
  containers,
  emSize: '16px',
  font,
  modals,
  palette,
  radioButton,
  reset,
  spacer,
  typography,
  zIndex
};
