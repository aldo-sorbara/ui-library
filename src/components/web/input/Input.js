/* stylelint-disable order/properties-alphabetical-order */
import React, { useState } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const suggestions = ['gmail.com', 'hotmail.com', 'yahoo.com', 'outlook.com'];
export const Input = props => {
  const [suggestion, setSuggestion] = useState(false);
  const [value, setValue] = useState();

  const {
    meta: { error, valid, touched },
    label,
    placeholder,
    descriptionTop,
    descriptionBottom,
    description,
    size,
    input,
    onChange,
    prefix,
    type,
    callback,
    data,
    handleValueChange
  } = props;

  /* istanbul ignore next */

  const handleChange = e => {
    if (handleValueChange) {
      e.target.value = handleValueChange(e.target.value);
    }
    if (onChange) {
      onChange(e);
    } else {
      input.onChange && input.onChange(e);
    }
    setValue(value);
  };
  /* istanbul ignore next */

  const handleClick = suggestion => {
    const suggestValue = `${value.split('@')[0]}@${suggestion}`;
    setValue(suggestValue);
    setSuggestion(false);
    callback && callback(suggestValue);
  };
  /* istanbul ignore next */

  const getSuggester = () => (
    <SuggestionContainer>
      {suggestions.map((suggestion, i) => (
        <Item
          onClick={() => handleClick(suggestion)}
          key={`${suggestion}-${i}`}
        >{`@${suggestion}`}</Item>
      ))}
    </SuggestionContainer>
  );

  /* istanbul ignore next */

  const handleKeyPress = e => {
    const inputValue = e.target.value.split('@')[1];
    setValue(e.target.value);
    switch (type) {
      case 'email':
        if (e.which === 50) {
          setSuggestion(true);
        } else {
          const isIncluded = suggestions.filter(suggestion =>
            suggestion.startsWith(inputValue)
          );
          !isIncluded.length ? setSuggestion(false) : setSuggestion(true);
        }
        break;
      default:
        return;
    }
  };
  return (
    <InputContainer>
      {(label || descriptionTop) && (
        <InputLabel>
          {label} {descriptionTop && getDescription(description)}
        </InputLabel>
      )}
      <InputBlock size={size}>
        {prefix && <Prefix size={size}>{prefix}</Prefix>}

        <InputText
          data-w={data}
          size={size}
          {...props}
          {...props.input}
          onKeyDown={handleKeyPress}
          onChange={handleChange}
          placeholder={placeholder}
        />
        {suggestion && getSuggester()}
        <InputStatus size={size} {...props.meta} />

        <InputBorder {...props} />
      </InputBlock>

      {!valid && touched ? (
        <InputError>{error || description}</InputError>
      ) : (
        descriptionBottom && getDescription(description)
      )}
    </InputContainer>
  );
};

Input.defaultProps = {
  meta: {
    error: null,
    visited: false,
    dirty: false,
    active: false,
    valid: false,
    touched: false
  },
  input: {},
  onChange: null,
  prefix: null,
  size: 'default',
  descriptionTop: false,
  descriptionBottom: false,
  label: null
};

Input.propTypes = {
  meta: PropTypes.object,
  size: PropTypes.string,
  input: PropTypes.object,
  onChange: PropTypes.func,
  handleValueChange: PropTypes.func,
  callback: PropTypes.func,
  descriptionTop: PropTypes.bool,
  descriptionBottom: PropTypes.bool,
  description: PropTypes.node,
  label: PropTypes.string,
  placeholder: PropTypes.string.isRequired,
  prefix: PropTypes.string,
  data: PropTypes.string.isRequired
};

const getDescription = description => (
  <div>
    <InputDescription>{description}</InputDescription>
  </div>
);

const Icon = props => {
  const { dirty, valid } = props;
  if (dirty) {
    if (valid) {
      return IconSuccess;
    }
    return IconError;
  }
  return null;
};

const IconSuccess = (
  <svg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 24 24'>
    <path d='M0 0h24v24H0z' fill='none' />
    <path
      fill='#55AE07'
      d='M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z'
    />
  </svg>
);

const IconError = (
  <svg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 24 24'>
    <path
      fill='#d0021b'
      d='M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z'
    />
    <path d='M0 0h24v24H0z' fill='none' />
  </svg>
);

const InputStatus = props => {
  const { size } = props;
  return (
    <IconContainer size={size}>
      <Icon {...props} />
    </IconContainer>
  );
};

const setInputBorderColor = props => {
  const { active, dirty, visited, error } = props.meta;
  const { secondary, error: errorColor, success, common } = props.theme.palette;

  if (active) {
    return secondary.default;
  } else if (visited && dirty) {
    if (error) {
      return errorColor.default;
    } else {
      return success.default;
    }
  } else {
    return common.text.default;
  }
};

const InputContainer = styled.div`
  width: 100%;
`;

const SuggestionContainer = styled.div`
  background: white;
  border-style: solid;
  border-width: 0 1px 1px 1px;
  box-shadow: 0 2px 3px 0 rgba(34, 36, 38, 0.15);
  max-height: 20rem;
  outline: 0;
  overflow-x: hidden;
  overflow-y: auto;
  position: absolute;
  transition: opacity 0.1s ease;
  top: 2rem;
  width: 100%;
  z-index: 9999;
`;

/* istanbul ignore next */

const Item = styled.div`
  background: white;
  border-top: none;
  box-shadow: none;
  cursor: pointer;
  display: block;
  font-size: 1rem;
  height: auto;
  line-height: 1;
  ${props => props.theme.typography.label.default};
  padding: 0.75rem 1rem;
  position: relative;
  text-align: left;
  text-transform: none;
  white-space: normal;
  word-wrap: normal;
  &:hover {
    background: ${props => props.theme.palette.secondary.light};
  }
`;

Item.defaultProps = {
  theme: {
    typography: {
      label: {
        default: {
          fontSize: '1rem',
          fontWeight: '700',
          lineHeight: '1.33'
        }
      }
    },
    palette: {
      secondary: {
        light: 'inherit'
      }
    }
  }
};

const Prefix = styled.span`
  ${props =>
    props.theme.typography.label[props.size] ||
    props.theme.typography.label.default};
  align-items: center;
  display: flex;
  font-size: inherit;
  font-weight: normal;
  padding-right: ${props => props.theme.spacer.xSmall};
  text-align: center;
  white-space: nowrap;
`;

const InputText = styled.input`
  ${props =>
    props.theme.typography.label[props.size] ||
    props.theme.typography.label.default};
  border: 0;
  flex: 1 1 auto;
  font-size: inherit;
  font-weight: normal;
  width: 1%;

  &:focus {
    ~ div {
      background-color: ${props => props.theme.palette.secondary.default};
      opacity: 1;
    }
    outline: none;
  }

  /* Change Autocomplete styles in Chrome */
  &:-webkit-autofill,
  &:-webkit-autofill:hover,
  &:-webkit-autofill:focus {
    background-color: transparent !important;
    -webkit-box-shadow: none !important;
    -webkit-text-fill-color: #000;
    transition: background-color 5000s ease-in-out 0s;
  }

  &::placeholder {
    color: ${props => props.theme.palette.common.disabled};
  }
`;

export const InputLabel = styled.label`
  display: block;
  line-height: 1;
  ${props => props.theme.typography.label.default};
  font-weight: 700;
  padding-bottom: ${props => props.theme.spacer.xSmall};
`;

export const InputDescription = styled.p`
  display: inline-block;
  line-height: 1;
  margin-bottom: 0;
  ${props => props.theme.typography.caption};
  font-size: 0.75rem;
  padding-top: 4px;
`;

const InputError = styled.p`
  color: ${props => props.theme.palette.error.default};
  display: inline-block;
  line-height: 1;
  margin-bottom: 0;
  ${props => props.theme.typography.description};
  font-size: 0.75rem;
`;

const InputBlock = styled.div`
  align-items: stretch;
  display: flex;
  flex-wrap: wrap;
  font-size: ${props =>
    props.theme.typography.input[props.size]
      ? props.theme.typography.input[props.size]['font-size']
      : props.theme.typography.input.default['font-size']};
  font-weight: 300;
  margin-bottom: ${props => props.theme.spacer.xSmall};
  ${props =>
    props.theme.spacer.input[props.size] || props.theme.spacer.input.default}
  position: relative;
  width: 100%;
`;
const InputBorder = styled.div`
  background-color: ${props => setInputBorderColor(props)};
  opacity: ${props =>
    setInputBorderColor(props) === props.theme.palette.common.text.default &&
    !props.meta.active
      ? 0.2
      : 1};
  bottom: 0;
  display: block;
  height: 1px;
  left: 0;
  position: absolute;
  right: 0;
  transition: background-color 0.2s ease-out;
  width: 100%;
`;

const IconContainer = styled.span`
  align-items: center;
  display: flex;
  font-size: inherit;
  ${props =>
    props.theme.typography.label[props.size] ||
    props.theme.typography.label.default};
  white-space: nowrap;
  & > svg {
    display: block;
    height: ${props =>
      props.theme.typography.input[props.size]
        ? props.theme.typography.input[props.size].iconSize
        : props.theme.typography.input.default.iconSize};
    width: ${props =>
      props.theme.typography.input[props.size]
        ? props.theme.typography.input[props.size].iconSize
        : props.theme.typography.input.default.iconSize};
  }
`;

export default Input;
