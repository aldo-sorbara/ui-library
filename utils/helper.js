import { shallow, mount } from 'enzyme';

import { ThemeConsumer } from 'styled-components';

export const shallowWithTheme = (children, theme) => {
  ThemeConsumer._currentValue = theme;
  return shallow(children);
};

export const mountWithTheme = (children, theme) => {
  ThemeConsumer._currentValue = theme;
  return mount(children);
};
