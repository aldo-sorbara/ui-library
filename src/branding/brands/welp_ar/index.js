import wording from './jsons/wording.json';
import theme from './theme';
import config from './jsons/config.json';
import feedback from './jsons/feedback.json';
import assets from './jsons/assets.json';
import palette from './jsons/palette.json';

export default {
  wording: { onboarding: wording },
  feedback: { onboarding: feedback },
  theme,
  config: { onboarding: config },
  typography: theme.font.fontfamily,
  assets: { onboarding: assets },
  palette
};
