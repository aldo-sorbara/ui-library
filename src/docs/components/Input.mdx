---
name: Input
menu: Components
route: /input
---

import { Playground, PropsTable } from 'docz';
import { Fragment } from 'react';
import Input from '../../components/web/input';
import { ThemeProvider } from 'styled-components';
import welp_ar from '../../branding/brands/welp_ar';
import GlobalStyle from '../../components/web/globalstyle';
import { styles } from '../../../utils/docz';

# Input

<PropsTable of={Input} />

## Basic usage

<Playground>
  <ThemeProvider theme={welp_ar.theme}>
    <>
      <GlobalStyle theme={welp_ar.theme} />
      <div style={{ margin: '40px' }}>
        <Input
          data='ingrese su celular'
          label='Ingrese su celular'
          name='validationcellphone'
          prefix='15'
          type='number'
          placeholder='42424242'
        />
      </div>
      <div style={{ margin: '40px' }}>
        <Input
          data='ingrese su celular'
          label='Ingrese su celular'
          name='validationcellphone'
          prefix='15'
          description='sin el 0 y sin el 11'
          descriptionBottom
          type='number'
          placeholder='42424242'
        />
      </div>
      <div style={{ margin: '40px' }}>
        <Input
          data='ingrese-celular'
          meta={{
            error: 'Numero invalido',
            dirty: true,
            touched: true,
            visited: true
          }}
          label='Ingrese su celular'
          name='validationcellphone'
          prefix='15'
          type='number'
          placeholder='42424242'
        />
      </div>
      <div style={{ margin: '40px' }}>
        <Input
          data='ingrese-celular'
          meta={{ valid: true, touched: true, visited: true, dirty: true }}
          label='Ingrese su celular'
          name='validationcellphone'
          prefix='15'
          type='number'
          placeholder='42424242'
          description='sin el 0 y sin el 11'
          descriptionBottom
        />
      </div>
      <div style={{ margin: '40px' }}>
        <Input
          data='ingrese su celular'
          label='Ingrese su celular'
          name='validationcellphone'
          prefix='15'
          description='sin el 0 y sin el 11'
          descriptionBottom
          type='number'
          placeholder='42424242'
          size='large'
        />
      </div>
      <div style={{ margin: '40px' }}>
        <Input
          data='ingrese su celular'
          label='Ingrese su celular'
          name='validationcellphone'
          prefix='15'
          description='sin el 0 y sin el 11'
          descriptionTop
          type='number'
          placeholder='42424242'
          size='xLarge'
        />
      </div>
    </>
  </ThemeProvider>
</Playground>

[Try it on Storybook!](./storybook/index.html?path=/story/input--with-prefix-label-and-placeholder)
