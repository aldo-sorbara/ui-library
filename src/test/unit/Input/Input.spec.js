import React from 'react';
import Input from '../../../components/web/input';
import renderer from 'react-test-renderer';
import { mountWithTheme } from '../../../../utils/helper';
import welp_ar from '../../../branding/brands/welp_ar';
import { ThemeProvider } from 'styled-components';
import 'jest-styled-components';

describe('Input component', () => {
  it('should render snapshot', () => {
    const component = renderer.create(
      <ThemeProvider theme={welp_ar.theme}>
        <Input
          data='ingrese su celular'
          label='Ingrese su celular'
          name='validationcellphone'
          prefix='15'
          type='number'
          placeholder='42424242'
        />
      </ThemeProvider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render an input with description', () => {
    const wrapper = mountWithTheme(
      <Input
        data='ingrese su celular'
        label='Ingrese su celular'
        name='validationcellphone'
        prefix='15'
        type='number'
        placeholder='42424242'
        description='test'
        descriptionBottom
      />,
      welp_ar.theme
    );
    expect(wrapper.props().descriptionBottom).toBe(true);
    expect(wrapper.props().descriptionTop).not.toBe(true);
  });
  it('should render an input with error', () => {
    const wrapper = mountWithTheme(
      <Input
        data='ingrese su celular'
        label='Ingrese su celular'
        name='validationcellphone'
        prefix='15'
        type='number'
        placeholder='42424242'
        description='test'
        descriptionBottom
        meta={{
          error: 'error',
          valid: false,
          touched: true,
          dirty: true,
          visited: true
        }}
      />,
      welp_ar.theme
    );
    expect(wrapper.props().meta).toEqual({
      valid: false,
      touched: true,
      dirty: true,
      visited: true,
      error: 'error'
    });
  });
  it('should render an input with description error', () => {
    const wrapper = mountWithTheme(
      <Input
        data='ingrese su celular'
        label='Ingrese su celular'
        name='validationcellphone'
        prefix='15'
        type='number'
        placeholder='42424242'
        description='test'
        descriptionBottom
        meta={{
          valid: false,
          touched: true,
          dirty: true,
          visited: true
        }}
      />,
      welp_ar.theme
    );
    expect(wrapper.props().meta).toEqual({
      valid: false,
      touched: true,
      dirty: true,
      visited: true
    });
  });
  it('should render an active input', () => {
    const wrapper = mountWithTheme(
      <Input
        data='ingrese su celular'
        label='Ingrese su celular'
        name='validationcellphone'
        prefix='15'
        type='number'
        placeholder='42424242'
        description='test'
        descriptionBottom
        meta={{ active: true }}
      />,
      welp_ar.theme
    );
    expect(wrapper.props().meta).toEqual({ active: true });
  });
  it('should render an icon success', () => {
    const wrapper = mountWithTheme(
      <Input
        data='ingrese su celular'
        label='Ingrese su celular'
        name='validationcellphone'
        prefix='15'
        type='number'
        placeholder='42424242'
        description='test'
        descriptionBottom
        meta={{ dirty: true, visited: true, valid: true }}
      />,
      welp_ar.theme
    );
    expect(wrapper.props().meta).toEqual({
      dirty: true,
      visited: true,
      valid: true
    });
  });
  it('should render an input with default size', () => {
    const wrapper = mountWithTheme(
      <Input
        data='ingrese su celular'
        label='Ingrese su celular'
        name='validationcellphone'
        prefix='15'
        type='number'
        placeholder='42424242'
        description='test'
        descriptionBottom
        size='test'
        meta={{ active: true }}
      />,
      welp_ar.theme
    );
    expect(wrapper.props().size).toBe('test');
    expect(
      wrapper
        .find('span')
        .at(0)
        .prop('size')
    ).toBe('test');
  });
  it('should render an input with descriptionTop', () => {
    const wrapper = mountWithTheme(
      <Input
        data='ingrese su celular'
        name='validationcellphone'
        prefix='15'
        type='number'
        placeholder='42424242'
        description='test'
        descriptionTop
        meta={{ active: true }}
      />,
      welp_ar.theme
    );
    expect(wrapper.props().description).toBe('test');
    expect(wrapper.props().descriptionTop).toBe(true);
  });
  it('should render an input with suggester', () => {
    const wrapper = mountWithTheme(
      <Input
        data='ingrese su celular'
        name='validationcellphone'
        type='email'
        placeholder='42424242'
        description='test'
        descriptionTop
        meta={{ active: true }}
        value='test@'
      />,
      welp_ar.theme
    );
    wrapper.find('input').simulate('keyDown', { which: 50 });
    expect(wrapper.props().value).toBe('test@');
  });
});
