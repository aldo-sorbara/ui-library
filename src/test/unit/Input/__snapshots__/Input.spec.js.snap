// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component should render snapshot 1`] = `
.c0 {
  width: 100%;
}

.c3 {
  font-size: 1rem;
  font-weight: 700;
  line-height: 1.33;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: inherit;
  font-weight: normal;
  padding-right: .25rem;
  text-align: center;
  white-space: nowrap;
}

.c4 {
  font-size: 1rem;
  font-weight: 700;
  line-height: 1.33;
  border: 0;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  font-size: inherit;
  font-weight: normal;
  width: 1%;
}

.c4:focus {
  outline: none;
}

.c4:focus ~ div {
  background-color: #c346f9;
  opacity: 1;
}

.c4:-webkit-autofill,
.c4:-webkit-autofill:hover,
.c4:-webkit-autofill:focus {
  background-color: transparent !important;
  -webkit-box-shadow: none !important;
  -webkit-text-fill-color: #000;
  -webkit-transition: background-color 5000s ease-in-out 0s;
  transition: background-color 5000s ease-in-out 0s;
}

.c4::-webkit-input-placeholder {
  color: #d5d8df;
}

.c4::-moz-placeholder {
  color: #d5d8df;
}

.c4:-ms-input-placeholder {
  color: #d5d8df;
}

.c4::placeholder {
  color: #d5d8df;
}

.c1 {
  display: block;
  line-height: 1;
  font-size: 1rem;
  font-weight: 700;
  line-height: 1.33;
  font-weight: 700;
  padding-bottom: .25rem;
}

.c2 {
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  font-size: 1rem;
  font-weight: 300;
  margin-bottom: .25rem;
  padding: .25rem .25rem .25rem 0;
  line-height: 1.5;
  position: relative;
  width: 100%;
}

.c6 {
  background-color: #000000;
  opacity: 0.2;
  bottom: 0;
  display: block;
  height: 1px;
  left: 0;
  position: absolute;
  right: 0;
  -webkit-transition: background-color 0.2s ease-out;
  transition: background-color 0.2s ease-out;
  width: 100%;
}

.c5 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: inherit;
  font-size: 1rem;
  font-weight: 700;
  line-height: 1.33;
  white-space: nowrap;
}

.c5 > svg {
  display: block;
  height: 1.125rem;
  width: 1.125rem;
}

<div
  className="c0"
>
  <label
    className="c1"
  >
    Ingrese su celular
     
  </label>
  <div
    className="c2"
    size="default"
  >
    <span
      className="c3"
      size="default"
    >
      15
    </span>
    <input
      className="c4"
      data="ingrese su celular"
      data-w="ingrese su celular"
      label="Ingrese su celular"
      name="validationcellphone"
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder="42424242"
      prefix="15"
      size="default"
      type="number"
    />
    <span
      className="c5"
      size="default"
    />
    <div
      className="c6"
      data="ingrese su celular"
      label="Ingrese su celular"
      name="validationcellphone"
      onChange={null}
      placeholder="42424242"
      prefix="15"
      size="default"
      type="number"
    />
  </div>
</div>
`;
